### PR TITLE
Support for MSC2457 logout_devices param for setPassword()

### PR DESCRIPTION
**Pull request by hughns**
---
Here's what your changelog entry will look like:

## ✨ Features
 * Support for MSC2457 logout_devices param for setPassword() ([\#2285](https://github.com/matrix-org/matrix-js-sdk/pull/2285)). Contributed by @hughns.

### CODE REVIEW

File: src/client.ts [hunk 1]
```diff
@@ -7815,15 +7815,35 @@ export class MatrixClient extends TypedEventEmitter<EmittedEvents, ClientEventHa
...
      * @return {Promise} Resolves: TODO
      * @return {module:http-api.MatrixError} Rejects: with an error response.
      */
-    public setPassword(authDict: any, newPassword: string, callback?: Callback): Promise<any> { // TODO: Types
+    public setPassword(
+        authDict: any,
+        newPassword: string,
+        logoutDevices: boolean,
+        callback?: Callback
+        ): Promise<any>;
```
Comment (t3chguy) [inline src/client.ts:7828]:
```suggestion
        callback?: Callback,
    ): Promise<any>;
```

File: src/client.ts [hunk 2]
```diff
@@ -7815,15 +7815,35 @@ export class MatrixClient extends TypedEventEmitter<EmittedEvents, ClientEventHa
...
+        newPassword: string,
+        logoutDevices?: Callback | boolean,
+        callback?: Callback,
+    ): Promise<any> { // TODO: Types
+        if (typeof logoutDevices !== 'boolean') {
+            logoutDevices = true;
+        }
+        if (typeof logoutDevices === 'function') {
+            callback = logoutDevices;
+        }
```
Comment (t3chguy) [inline src/client.ts:7840]: I think these might be backwards, clobbering the callback to be `true` in the backwards compat case

**Review by t3chguy — CHANGES_REQUESTED** [review]

File: src/client.ts (hunk 2 above)
Comment (hughns) [reply src/client.ts]: Well spotted, thank you.

**Review by t3chguy — APPROVED** [review]

**Comment by t3chguy** [comment]
Optional: could we get test coverage over the backwards compatible signature?

**Comment by hughns** [comment]
@t3chguy i added tests trying to follow existing patterns.

In doing so I noted that it would be better to make the default behaviour actually mimic the current behaviour. Specifically making it so that if the existing setPassword signature is used then the `logout_devices` param is not specified in the `POST` request.

**Review by t3chguy — APPROVED** [review]
:shipit:

**Pull request merged**